scattering/van_hove.py

```diff
@@ -275,3 +283,91 @@ def compute_partial_van_hove(
         g_r_t += g_r_t_frame
 
     return r, g_r_t
+
+
+def get_unique_atoms(trj):
+    """Get mdtraj.Atom objects with unique `name`
+
+    Parameters
+    ----------
+    trj : MDTraj.trajectory
+        trajectory object of system
+
+    Returns
+    -------
+    unique_atoms : list
+        List of unique atoms in trajectory
+
+    """
+    seen_atoms = set()
+    unique_atoms = [
+        seen_atoms.add(atom.name) or atom
+        for atom in trj.topology.atoms
+        if atom.name not in seen_atoms
```

Comment:
This all looks good to me. The only comment I have is that I am unsure how this expression works, though I suspect I am just unfamiliar with how python handles boolean expressions.

Reply:
@lisankim0321 may need to clarify.  But I think what's happening is that `seen_atoms.add(atom.name) or atom` is done to both add to the `seen_atoms` set while grabbing `atom` in the list comprehension.  The `or` statement is used I think because `seen_atoms.add(atom.name)` will always return `False` and `atom` will always return `True`.

I think this would be readable however if the list comprehension was broken up.  You should be able to grab the `seen_atoms` with `seen_atoms = set([i.name for I in trj.topology.atoms])`